### PR TITLE
ath79: use u-boot,env

### PR DESCRIPTION
**Pull request by neheb**
Avoids having to handle reading in userspace.

### CODE REVIEW

File: package/boot/uboot-envtools/files/ath79 [hunk 1]
```diff
@@ -11,170 +11,6 @@ touch /etc/config/ubootenv
...
-	;;
-sophos,ap15|\
-sophos,ap55|\
-sophos,ap55c|\
-sophos,ap100|\
-sophos,ap100c)
-	ubootenv_add_uci_config "/dev/mtd1" "0x0" "0x1000" "0x10000"
-	;;
-wallys,dr531)
-	ubootenv_add_uci_config "/dev/mtd1" "0x0" "0xf800" "0x10000"
```
Comment (DragonBluep) [inline package/boot/uboot-envtools/files/ath79:169]: The same condition as glinet,gl-ar150.

File: target/linux/ath79/dts/qca9531_letv_lba-047-ch.dts [hunk 2]
```diff
@@ -91,6 +91,7 @@
 			};
 
 			partition@40000 {
+				compatible = "u-boot,env";
 				label = "u-boot-env";
```
Comment (DragonBluep) [inline target/linux/ath79/dts/qca9531_letv_lba-047-ch.dts:95]: It seems that you convert them by a script. This device doesn't have uboot-env config. Please also check other devices.

**Review by DragonBluep — COMMENTED** [review]
Thanks for your hard work.

File: target/linux/ath79/dts/ar9331_arduino_yun.dts [hunk 3]
```diff
@@ -167,6 +168,7 @@
 			};
 
 			partition@fe0000 {
+				compatible = "u-boot,env";
 				label = "nvram";
 				reg = <0xfe0000 0x10000>;
 			};
```
Comment (musashino205) [inline target/linux/ath79/dts/ar9331_arduino_yun.dts:174]: Is this partition compatible with `u-boot,env`?

File: target/linux/ath79/dts/ar9344_atheros_db120.dts [hunk 4]
```diff
@@ -133,6 +134,7 @@
 			};
 
 			partition@7e0000 {
+				compatible = "u-boot,env";
 				label = "nvram";
 				reg = <0x7e0000 0x010000>;
 			};
```
Comment (musashino205) [inline target/linux/ath79/dts/ar9344_atheros_db120.dts:140]: ditto

**Pull request closed without merging**